frontend/src/app/models/base-filter.ts

```diff
@@ -0,0 +1,195 @@
+import { ParamMap } from '@angular/router';
+import * as _ from 'lodash';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+/**
+ * Describes shared attributes and properties for filters.
+ */
+export interface FilterAPI<FilterData = any> {
```

Comment:
Perhaps a better name would be FilterInterface? `API` sounds as though this interacts with the backend.